Content.Shared/Mind/SharedMindSystem.cs

```diff
@@ -294,6 +294,33 @@ public bool TryRemoveObjective(EntityUid mindId, MindComponent mind, int index)
         return true;
     }
 
+    public bool TryGetObjectiveComp<T>(EntityUid uid, [NotNullWhen(true)] out T? objective) where T : Component
```

Comment:
a thing i just saw is wont this have conflicting signature:
- call with (uid, out var obj) it can pick either this or the other with mind as null
- call with (uid, out var obj, mind) it knows its the mind one

either rename one of them or make mindcomponent not default to null